app/browser/api/userModel.js

```diff
@@ -0,0 +1,325 @@
+  // on the other side, return early until these are populated
+  setImmediate(function () {
+    matrixData = um.getMatrixDataSync()
+    priorData = um.getPriorDataSync()
+    sampleAdFeed = um.getSampleAdFeed()
+  })
+
+  retrieveSSID()
+
+  return state
+}
+
+const tabUpdate = (state, action) => {
+  // nothing but update the ums for now
+  state = userModelState.setLastUserActivity(state)
+  return state
+}
+
+/* these two are pretty similar, userAction presently unused but maybe needed */
+const userAction = (state) => {
+  state = userModelState.setUserActivity()
+  return state
+}
+
+const removeHistorySite = (state, action) => {
+  // check to see how ledger removes history
+  // first need to establish site classification DB in userModelState
+
+  // blow it all away for now
+  state = userModelState.removeAllHistory(state)
+  return state
+}
+
+const removeAllHistory = (state) => {
+  state = userModelState.removeAllHistory(state)
+  return state
+}
+
+const saveCachedInfo = (state) => {
+  // writes stuff to leveldb
+  return state
+}
+
+const testShoppingData = (state, url) => {
+  const hostname = urlUtil.getHostname(url)
+  const lastShopState = userModelState.getSearchState(state)
+
+  if (hostname === 'amazon.com') {
+    const score = 1.0   // eventually this will be more sophisticated than if(), but amazon is always a shopping destination
+    state = userModelState.flagShoppingState(state, url, score)
+  } else if (hostname !== 'amazon.com' && lastShopState) {
+    state = userModelState.unFlagShoppingState(state)
+  }
+  return state
+}
+
+const testSearchState = (state, url) => {
+  const hostname = urlUtil.getHostname(url)
+  const lastSearchState = userModelState.getSearchState(state)
+  if (hostname === 'google.com') {
+    const score = 1.0  // eventually this will be more sophisticated than if(), but google is always a search destination
+    state = userModelState.flagSearchState(state, url, score)
+  } else if (hostname !== 'google.com' && lastSearchState) {
+    state = userModelState.unFlagSearchState(state, url)
+  }
+  return state
+}
+
+const recordUnIdle = (state) => {
+  state = userModelState.setLastUserIdleStopTime(state)
+  return state
+}
+
+function cleanLines (x) {
+  if (x == null) {
+    return []
+  }
+
+  return x
+    .map(x => x.split(/\s+/)) // split each: ['the quick', 'when in'] -> [['the', 'quick'], ['when', 'in']]
+    .reduce((x, y) => x.concat(y), []) // flatten: [[a,b], [c,d]] -> [a, b, c, d]
+    .map(x => x.toLowerCase().trim())
+}
+
+function randomKey (dictionary) {
+  const keys = Object.keys(dictionary)
+  return keys[keys.length * Math.random() << 0]
+}
+
+const goAheadAndShowTheAd = (windowId, notificationTitle, notificationText, notificationUrl) => {
+  appActions.nativeNotificationCreate(
+    windowId,
+    {
+      title: notificationTitle,
+      message: notificationText,
+      icon: path.join(__dirname, '../../../img/BAT_icon.png'),
```

Comment:
are you sure this works in packaged builds?

Reply:
i believe so, we are seeing it work now in the current RC under private test.